File: dash_docs/chapters/dash_core_components/Button/examples/button.R
library(dash)
library(dashHtmlComponents)

app <- Dash$new()

app$layout(
  htmlDiv(
    list(
      htmlDiv(dccInput(id="input-box", type="text")),
      htmlButton("Submit", id="button"),
      htmlDiv(id="output-container-button",
              children="Enter a value and press submit")
    )
  )
)

app$callback(
  output = list(id = "output-container-button", property = 'children'),
  params=list(input(id = "button", property = "n_clicks"),
              input(id = "input-box", property = "value")),
  function(n_clicks, value) {
    object_as_string <- capture.output(dput(n_clicks))
    if (is.list(n_clicks)) {
      sprintf("'n_clicks' is of length %d, with the value '%s'", length(n_clicks), object_as_string)
    } else {
      sprintf("'n_clicks' is of length %d, you have clicked the button %d time(s), and the value is '%s'", length(n_clicks), n_clicks, value)
    }
  }
)

app$run_server()

File: dash_docs/chapters/dash_html_components/Button/examples/button.py
import dash
import dash_html_components as html
import dash_core_components as dcc

external_stylesheets = ['https://codepen.io/chriddyp/pen/bWLwgP.css']

app = dash.Dash(__name__, external_stylesheets=external_stylesheets)
app.layout = html.Div([
    html.Div(dcc.Input(id='submit-input-box', type='text')),
    html.Button('Submit', id='submit-button-example'),
    html.Div(id='submit-output-button',
             children='Enter a value and press submit')
])


@app.callback(
    dash.dependencies.Output('submit-output-button', 'children'),
    [dash.dependencies.Input('submit-button-example', 'n_clicks')],
    [dash.dependencies.State('submit-input-box', 'value')])
def update_output(n_clicks, value):
    return 'The input value was "{}" and the button has been clicked {} times'.format(
        value,
        n_clicks
    )


if __name__ == '__main__':
    app.run_server(debug=True)
